src/statisticalmodel.jl

```diff
@@ -0,0 +1,243 @@
+"""
+    StatisticalModel
+
+Abstract supertype for all statistical models.
+"""
+abstract type StatisticalModel end
+
+"""
+    coef(model::StatisticalModel)
+
+Return the coefficients of the model.
+"""
+function coef end
+
+"""
+    coefnames(model::StatisticalModel)
+
+Return the names of the coefficients.
+"""
+function coefnames end
+
+"""
+    coeftable(model::StatisticalModel; level::Real=0.95)
+
+Return a table with coefficients and related statistics of the model.
+`level` determines the level for confidence intervals (by default, 95%).
+
+The returned `CoefTable` object implements the
+[Tables.jl](https://github.com/JuliaData/Tables.jl/) interface, and can be
+converted e.g. to a `DataFrame` via `using DataFrames; DataFrame(coeftable(model))`.
+"""
+function coeftable end
+
+"""
+    confint(model::StatisticalModel; level::Real=0.95)
+
+Compute confidence intervals for coefficients, with confidence level `level` (by default 95%).
+"""
+function confint end
+
+"""
+    deviance(model::StatisticalModel)
+
+Return the deviance of the model relative to a reference, which is usually when applicable
+the saturated model. It is equal, *up to a constant*, to ``-2 \\log L``, with ``L``
+the likelihood of the model.
+"""
+function deviance end
+
+"""
+    islinear(model::StatisticalModel)
+
+Indicate whether the model is linear.
+"""
+function islinear end
+
+"""
+    nulldeviance(model::StatisticalModel)
+
+Return the deviance of the null model, that is the one including only the intercept.
+"""
+function nulldeviance end
+
+"""
+    loglikelihood(model::StatisticalModel)
+
+Return the log-likelihood of the model.
+"""
+function loglikelihood end
+
+"""
+    loglikelihood(model::StatisticalModel)
+
+Return the log-likelihood of the null model corresponding to `model`.
+This is usually the model containing only the intercept.
+"""
+function nullloglikelihood end
+
+"""
+    loglikelihood(model::StatisticalModel, ::Colon)
+
+Return a vector of each observation's contribution to the log-likelihood of the model.
+In other words, this is the vector of the pointwise log-likelihood contributions.
+
+In general, `sum(loglikehood(model, :)) == loglikelihood(model)`.
+"""
+function loglikelihood end
```

Comment:
Does it work to have two stubs with different docstrings? Or should the two docstrings just be merged?

Reply:
Well, three actually:)

Reply:
Yeah there are duplicates after deleting the errors... I pretty much automated that and didn't notice the dups.

Reply:
And `nullloglikelyhood` seems to have always had that error